modules/integration_aws-elasticsearch/detectors-elasticsearch.tf

```diff
@@ -80,8 +80,10 @@ resource "signalfx_detector" "free_space" {
   }
 
   program_text = <<-EOF
-    free = data('FreeStorageSpace', filter=filter('namespace', 'AWS/ES') and filter('stat', 'lower') and filter('NodeId', '*') and ${module.filtering.signalflow})${var.free_space_aggregation_function}${var.free_space_transformation_function}
-    signal = free.scale(0.001).publish('signal')
+    data_node_free_storage   = data('FreeStorageSpace',       filter=filter('namespace', 'AWS/ES') and filter('stat', 'lower') and filter('NodeId', '*') and ${module.filtering.signalflow})${var.free_space_aggregation_function}${var.free_space_transformation_function}.scale(0.001)
+    warm_node_free_storage   = data('WarmFreeStorageSpace',   filter=filter('namespace', 'AWS/ES') and filter('stat', 'lower') and filter('NodeId', '*') and ${module.filtering.signalflow})${var.free_space_aggregation_function}${var.free_space_transformation_function}.scale(0.001)
```

Comment:
UltraWarm storage is radically different from classic storage (i.e. S3 vs EBS).

AWS documentation recommends different alert rules:

> Alarm: FreeStorageSpace minimum is <= 20480 for 1 minute, 1 consecutive time
> Issue: A node in your cluster is down to 20 GiB of free storage space. See Lack of available storage space. This value is in MiB, so rather than 20480, we recommend setting it to 25% of the storage space for each node.

> Alarm: WarmFreeStorageSpace minimum is <= 10240 for 1 minute, 1 consecutive time
> Issue: An UltraWarm node in your cluster is down to 10 GiB of free storage space. See Lack of available storage space. This value is in MiB, so rather than 10240, we recommend setting it to 10% of the storage space for each UltraWarm node.

I believe we should create a separate detector for UltraWarm nodes.

And maybe adjust the threshold for classic storage.

Reply:
done